lib/init.coffee

```diff
@@ -1,8 +1,61 @@
+{BufferedProcess, CompositeDisposable} = require 'atom'
+path = require 'path'
+helpers = require 'atom-linter'
+
 module.exports =
   config:
     javaExecutablePath:
       type: 'string'
-      default: ''
+      title: 'Path to the javac executable'
+      default: 'javac'
 
   activate: ->
-    console.log 'activate linter-javac'
+    @subscriptions = new CompositeDisposable
+    @subscriptions.add atom.config.observe 'linter-javac.javaExecutablePath',
+      (newValue) =>
+        @javaExecutablePath = newValue
+
+  deactivate: ->
+    @subscriptions.dispose()
+
+  provideLinter: ->
+    grammarScopes: ['source.java']
+    scope: 'project'
+    lintOnFly: false       # Only lint on save
+    lint: (textEditor) =>
+      filePath = textEditor.getPath()
+      wd = path.dirname filePath
+      # Use the text editor's working directory as the classpath.
+      #  TODO: Make the classpath user configurable.
+      args = ['-Xlint:all', '-cp', wd, filePath]
+      messages = []
+      helpers.exec(@javaExecutablePath, args, {throwOnStdErr: false, stream: 'stderr'})
+        .then (val) => return @parse(val, textEditor)
+        .catch (val) =>
```

Comment:
You don't actually have to catch it here, the last time I checked, `atom-linter` throws this and `linter` catches